mlx/backend/common/default_primitives.cpp

```diff
@@ -190,4 +224,140 @@ void AddMM::eval_cpu(const std::vector<array>& inputs, array& out) {
   return matmul_common_general(inputs[0], inputs[1], out, alpha_, beta_);
 }
 
+void BlockMaskedMM::eval_cpu(const std::vector<array>& inputs, array& out) {
+  if (out.dtype() != float32) {
+    throw std::runtime_error(
+        "[BlockMaskedMM::eval_cpu] Currently only supports float32.");
+  }
+  out.set_data(allocator::malloc_or_wait(out.nbytes()));
+
+  auto& a_pre = inputs[0];
+  auto& b_pre = inputs[1];
+  auto& out_mask = inputs[2];
+
+  auto check_transpose = [](const array& arr, bool do_copy) {
```

Comment:
The `do_copy` flag is unused unless I miss something. Otherwise this function is the same as the `check_transpose` in `matmul_common_general`.  Probably worth refactoring it to a standalone helper since it doesn't save that much as a closure.

Reply:
So the `do_copy` flag plays a role when we have operand masks 
In which case a copy is always forced so that we can mask out the inputs inlace before sending them into the blas germ

Reply:
Ok.. but I don't see the call to `check_transpose` with `do_copy = true` anywhere. Is it missing or am I missing it?

Reply:
👍  Sorry I missed that

Reply:
It's in there 
```python
  bool has_op_mask = inputs.size() > 3;
  auto [a_transposed, lda, a] = check_transpose(a_pre, has_op_mask);
  auto [b_transposed, ldb, b] = check_transpose(b_pre, has_op_mask);
```